kotlinx-coroutines-core/jvm/src/scheduling/WorkQueue.kt

```diff
@@ -100,41 +101,37 @@ internal class WorkQueue {
     }
 
     /**
-     * Tries stealing from [victim] queue into this queue.
+     * Tries stealing from this queue into the [stolenTaskRef] argument.
      *
      * Returns [NOTHING_TO_STEAL] if queue has nothing to steal, [TASK_STOLEN] if at least task was stolen
      * or positive value of how many nanoseconds should pass until the head of this queue will be available to steal.
      */
-    fun tryStealFrom(victim: WorkQueue): Long {
-        assert { bufferSize == 0 }
-        val task  = victim.pollBuffer()
+    fun trySteal(stolenTaskRef: ObjectRef<Task?>): Long {
```

Comment:
The names no longer reflect who's the receiver and who's doing the action. The semantics of "I steal (from myself) (into this bag that was given to me)" is not really intuitive!
* `tryGift`? Semantically clear, but "stealing" is a well-established term.
* `tryStealFromThisInto`? Wordy but much less confusing.
* Make it a separate static function, like `trySteal(victim, stolenTaskRef)`.
* ...

Reply:
Maybe `tryStealInto()`? `victim.tryStealInto(stolenTaskSlot)` seems quite straightforward

Reply:
"I steal (into this bag)" begs the question, from whom do you steal? The answer is, from yourself, which is just not obvious. When I see `X.y()`, I always read this as "`X`, I order you to `y`".

Reply:
I agree with the general reasoning. That's why I added context (receiver name), as this is the method used in a very specific manner on call sites we control, including the receiver's name.

I'm reluctant to both adding a companion (so the static method has FQN name like `WorkQueue.trySteal(victim, into)`) and to giving up on `steal` terminology

Reply:
Eh , ok. I think this warrants a comment though, in the vein of
"Counterintuitively, the queue from which a task is stolen is [this], so we are stealing from ourselves here. The reason for the name is, *stealing* is an established concept in concurrency".